capstone/src/main/java/com/google/step/clients/MetricsClient.java

```diff
@@ -0,0 +1,288 @@
+                Arrays.asList(
+                        new FilterPredicate("restaurantKey", Query.FilterOperator.EQUAL, restaurantKey),
+                        new FilterPredicate("year", Query.FilterOperator.EQUAL, year),
+                        new FilterPredicate("week", Query.FilterOperator.EQUAL, week)
+                ));
+
+        Query query = new Query("PageViews").setFilter(compositeFilter);
+        PreparedQuery preparedQuery = dataStore.prepare(query);
+        Entity results = preparedQuery.asSingleEntity();
+
+        // If there are no entities of that restaurant, add one with a count of 0.
+        if (results == null) {
+            Transaction transaction = dataStore.beginTransaction();
+            try {
+                Entity entity = new Entity("PageViews");
+                entity.setProperty("restaurantKey", restaurantKey);
+                entity.setProperty("restaurantName", restaurantName);
+                entity.setProperty("year", year);
+                entity.setProperty("week", week);
+                entity.setProperty("count", 0);
+                dataStore.put(transaction, entity);
+
+                // Set the count for the return map as 0, since the entity doesn't exist.
+                resultMap.put("count", 0);
+                transaction.commit();
+            } finally {
+                if (transaction.isActive()) {
+                    transaction.rollback();
+                }
+            }
+        } else {
+            // If the results didn't come back null, give the count.
+            resultMap.put("count", results.getProperty("count"));
+        }
+
+        resultMap.put("year", year);
+        resultMap.put("week", week);
+
+        return resultMap;
+    }
+
+    /**
+     * Gets the weekly pageViews for the entire year of a specific restaurant.
+     * @param year we are looking for
+     * @param restaurantKey of restaurant we want the data.
+     * @return list of maps in the following way: List<Map<String, Object>>
+     *          [
+     *                 {
+     *                      "year": <String, the year number>,
+     *                      "week": <String, this week's number in relation to the year.>,
+     *                      "count": <String, amount of page views the restaurant had.>
+     *                  },
+     *                  ...,
+     *          ]
+     */
+    public List<Map<String, Object>> getYearRestaurantPageViews(int year, String restaurantKey) {
+
+        CompositeFilter compositeFilter = new CompositeFilter(Query.CompositeFilterOperator.AND,
+                Arrays.asList(
+                        new FilterPredicate("restaurantKey", Query.FilterOperator.EQUAL, restaurantKey),
+                        new FilterPredicate("year", Query.FilterOperator.EQUAL, year)
+                ));
+        Query query = new Query("PageViews").setFilter(compositeFilter).addSort("week", Query.SortDirection.ASCENDING);
```

Comment:
This is good; you should add it to the documentation. We should just always guarantee that it will be sorted in ascending order, and let that be the contract of using the metrics client.

Reply:
Now it is still sorted, but only within each list inside every RestaurantPageViews class.

Reply:
Nit: Can you add the info about it being sorted only within each list inside RestaurantPageViews? Also please specify the order of the sort in the comment (e.g. ascending or descending).